asset-transfer-secured-agreement/application-javascript/app.js

```diff
@@ -0,0 +1,563 @@
+				// the actual peers that may be active at any given time.
+				const asset_properties = {
+					object_type: 'asset_properties',
+					asset_id: assetKey,
+					color: 'blue',
+					size: 35,
+					salt: Buffer.from(randomNumber.toString()).toString('hex')
+				};
+				const asset_properties_string = JSON.stringify(asset_properties);
+				console.log(`${GREEN}--> Submit Transaction: CreateAsset, ${assetKey} as Org1 - endorsed by Org1${RESET}`);
+				console.log(`${asset_properties_string}`);
+				transaction = contractOrg1.createTransaction('CreateAsset');
+				transaction.setEndorsingOrganizations(org1);
+				transaction.setTransient({
+					asset_properties: Buffer.from(asset_properties_string)
+				});
+				await transaction.submit(assetKey, `Asset ${assetKey} owned by ${org1} is not for sale`);
+				console.log(`*** Result: committed, asset ${assetKey} is owned by Org1`);
+			} catch (createError) {
+				console.log(`${RED}*** Failed: CreateAsset - ${createError}${RESET}`);
+			}
+
+			// read the public details by both orgs
+			await readAsset(assetKey, org1, contractOrg1, contractOrg2);
+			// Org1 should be able to read the private data details of this asset
+			await readPrivateAsset(assetKey, org1, contractOrg1);
+			// Org2 is not the owner and does not have the private details, this should fail
+			await readPrivateAsset(assetKey, org2, contractOrg2);
+
+			try {
+				// This is an update to the public state and requires only the owner to endorse.
+				console.log(`${GREEN}--> Submit Transaction: ChangePublicDescription ${assetKey}, as Org1 - endorse by Org1${RESET}`);
+				transaction = contractOrg1.createTransaction('ChangePublicDescription');
+				transaction.setEndorsingOrganizations(org1);
+				await transaction.submit(assetKey, `Asset ${assetKey} owned by ${org1} is for sale`);
+				console.log(`*** Result: committed, asset ${assetKey} is now for sale by Org1`);
+			} catch (updateError) {
+				console.log(`${RED}*** Failed: ChangePublicDescription - ${updateError}${RESET}`);
+			}
+
+			// read the public details by both orgs
+			await readAsset(assetKey, org1, contractOrg1, contractOrg2);
+
+			try {
+				// This is an update to the public state and requires only the owner to endorse.
+				// Since the Org2 is not the owner, this will fail
+				console.log(`${GREEN}--> Submit Transaction: ChangePublicDescription ${assetKey}, as Org2 - endorse by Org2${RESET}`);
+				transaction = contractOrg2.createTransaction('ChangePublicDescription');
+				transaction.setEndorsingOrganizations(org2);
+				await transaction.submit(assetKey, `Asset ${assetKey} owned by ${org2} is NOT for sale`);
+				console.log(`${RESET}*** Failed: Org2 is not the owner and this should have failed${RESET}`);
+			} catch (updateError) {
+				console.log(`*** Success: ChangePublicDescription has failed endorsememnt by Org2 - ${updateError}`);
+			}
+
+			try {
+				// This is an update to the public state and requires only the owner to endorse.
```

Comment:
This one is a little confusing, because you say owner must endorse, you send the endorsement to org1 (the owner), and yet it fails.
So highlight that the CLIENT in this scenario is Org2 and therefore it will fail.

Reply:
done